### PR TITLE
[fix] fix illegal mem access and clean up triton attention backend

### PR DESCRIPTION
**Pull request by Alcanderian**
## Motivation

Clean up and verify issue at https://github.com/sgl-project/sglang/pull/4553#issuecomment-2735283306

## Modifications

1. Extract `attn_lse` from `attn_logits` and add `ForwardMetadata` class for clearerdefinition.
2. Fix `get_num_kv_splits` with MTP

## Checklist

- [x] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

**Comment by merrymercy** [comment]
https://github.com/sgl-project/sglang/pull/4553#discussion_r2003682754

**Comment by zhyncs** [comment]
three gold bro is great!
